sql-plugin/src/main/330db/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

```diff
@@ -74,8 +76,24 @@ object SparkShimImpl extends Spark321PlusDBShims {
     parent match {
       case bhj: GpuBroadcastHashJoinExec =>
         shuffle.shuffleOrigin.equals(EXECUTOR_BROADCAST)
+      case bnlj: GpuBroadcastNestedLoopJoinExec =>
+        shuffle.shuffleOrigin.equals(EXECUTOR_BROADCAST)
+      case _ => false
+    }
+  }
+
+  override def addTransitionalShuffleIfNeeded(c2r: GpuColumnarToRowExec): SparkPlan = {
+    val sqse = c2r.find {
+      case e: ShuffleQueryStageExec => true
       case _ => false
     }
+    val plan = GpuTransitionOverrides.getNonQueryStagePlan(sqse.get)
```

Comment:
What if sqse is None?  This method takes a columnar to row exec which definitely does not necessitate a shuffle query stage exec is somewhere beneath it in the plan.

It looks like this is always called in a context where the shuffleQueryStageExec is already known, seems like it would be better if we just passed it as another argument here rather than needing to find it.  Either that or this should be able to match on the c2r.child node being ShuffleQueryStageExec and then go from there.  Using `find` seems off here.